### PR TITLE
chor(release-links): Add 3.11.0 alpha3 release links to downloads page

### PR DESCRIPTION
**Pull request by andromaqui**

### CODE REVIEW

File: enterprise/content/download.md [hunk 1]
```diff
@@ -3356,6 +3356,16 @@ Please note that Optimize alpha releases are not suitable for production environ
     </tr>
   </thead>
   <tbody>
+    <tr class="well">
+        <td>3.11.0-alpha3</td>
+        <td>7.18+, 7.19+</td>
+        <td>7.16.2+, 7.17.0+, 8.7.0+, 8.8.0+</td>
+        <td>13.06.2023</td>
```
Comment (PHWaechtler) [inline enterprise/content/download.md:3363]: This should be today's date

File: enterprise/content/download.md [hunk 2]
```diff
@@ -3356,6 +3356,16 @@ Please note that Optimize alpha releases are not suitable for production environ
...
+        <td>
+          <a class="btn btn-sm btn-default optimize-download" href="https://downloads.camunda.cloud/enterprise-release/optimize/3.11.0-alpha2/camunda-optimize-3.11.0-alpha2-production.zip">zip</a>&nbsp; 
+          <a class="btn btn-sm btn-default optimize-download" href="https://downloads.camunda.cloud/enterprise-release/optimize/3.11.0-alpha2/camunda-optimize-3.11.0-alpha2-production.tar.gz">tar.gz</a>
+        </td>
```
Comment (PHWaechtler) [inline enterprise/content/download.md:3367]: These still point to alpha2

File: enterprise/content/download.md [hunk 3]
```diff
@@ -3356,6 +3356,16 @@ Please note that Optimize alpha releases are not suitable for production environ
     </tr>
   </thead>
   <tbody>
+    <tr class="well">
+        <td>3.11.0-alpha3</td>
+        <td>7.18+, 7.19+</td>
+        <td>7.16.2+, 7.17.0+, 8.7.0+, 8.8.0+</td>
```
Comment (PHWaechtler) [inline enterprise/content/download.md:3362]: I think these aren't up to date, see here: https://github.com/camunda/camunda-optimize/blob/master/backend/src/main/java/org/camunda/optimize/service/util/ESVersionChecker.java#L31-L41 also lists versions below 7.16.2

**Review by PHWaechtler — CHANGES_REQUESTED** [review]
Some changes required before merging

File: enterprise/content/download.md (hunk 3 above)
Comment (andromaqui) [reply enterprise/content/download.md:3362]: I think this is for testing purposes but we have officially dropped the support, let me ask Josh

**Review by PHWaechtler — APPROVED** [review]
Lgtm

**Pull request merged**